### ISSUE TITLE
Question of Generator Loss of MaliGAN

### ISSUE THREAD

**Issue by cyk1337** [geek-ai/Texygen#47]
Is the generator loss of MaliGAN correct? It should be:

![image](https://user-images.githubusercontent.com/13767887/91018741-8867c500-e622-11ea-8988-c6bd96660982.png)

[https://github.com/geek-ai/Texygen/blob/3104e22ac75f3cc2070da2bf5e2da6d2bef149ad/models/maligan_basic/MaliganGenerator.py#L112](https://github.com/geek-ai/Texygen/blob/3104e22ac75f3cc2070da2bf5e2da6d2bef149ad/models/maligan_basic/MaliganGenerator.py#L112)
```python
        self.g_loss = -tf.reduce_sum(
            tf.reduce_sum(
                tf.one_hot(tf.to_int32(tf.reshape(self.x, [-1])), self.num_vocabulary, 1.0, 0.0) * tf.log(
                    tf.clip_by_value(tf.reshape(self.g_predictions, [-1, self.num_vocabulary]), 1e-20, 1.0)
                ), 1) * tf.reshape(self.rewards, [-1])
        )
```
